tests/app/main/test_session_timeout_warning.py

```diff
@@ -0,0 +1,131 @@
+            data_api_client.get_supplier_framework_info.return_value = \
+                SupplierFrameworkStub(slug=framework).single_result_response()
+            yield data_api_client
+
+    def test_framework_declaration_forms_have_session_timeout_warning(self, form, framework):
+        doc = html.fromstring(
+            self.client.get(f"/suppliers/frameworks/{framework}/declaration/edit/{form}").data
+        )
+
+        assert doc.xpath(".//div[@id='session-timeout-warning']")
+
+    def test_session_timeout_is_one_hour_in_future(self, form, framework, display_time):
+        with mock.patch("app.main.views.frameworks.render_template") as render_template:
+            render_template.return_value = ""
+
+            self.client.get(f"/suppliers/frameworks/{framework}/declaration/edit/{form}")
+
+            assert render_template.call_args[1]["session_timeout"] == display_time
+
+
+@pytest.mark.parametrize(
+    "framework",
+    (
+        "g-cloud-10",
+        "g-cloud-11",
+    )
+)
+@pytest.mark.parametrize(
+    "lot",
+    (
+        "cloud-software",
+        "cloud-support",
+    )
+)
+@pytest.mark.parametrize(
+    "form",
+    (
+        "about-your-service/service-categories",
+        "about-your-service/service-description",
+        "documents/service-definition-document-url",
+        "service-features-and-benefits/service-features-and-benefits",
+    )
+)
+class TestServiceSubmissionSessionTimeoutWarning(TestSessionTimeoutWarning):
+
+    @pytest.fixture(autouse=True)
+    def data_api_client(self, framework, lot):
+        self.login()
+        with mock.patch("app.main.views.services.data_api_client", autospec=True) as data_api_client:
+            data_api_client.get_framework.return_value = FrameworkStub(slug=framework).single_result_response()
+            data_api_client.get_supplier_framework_info.return_value = \
+                SupplierFrameworkStub(slug=framework).single_result_response()
+            data_api_client.get_draft_service.return_value = DraftServiceStub(
+                framework_slug=framework,
+                lot_slug=lot,
+                supplier_id=1234,
+            ).single_result_response()
+            yield data_api_client
+
+    def test_service_submission_forms_have_session_timeout_warning(self, form, framework, lot):
+        doc = html.fromstring(
+            self.client.get(f"/suppliers/frameworks/{framework}/submissions/{lot}/1/edit/{form}").data
+        )
+
+        assert doc.xpath(".//div[@id='session-timeout-warning']")
+
+    def test_session_timeout_is_one_hour_in_future(self, form, framework, lot, display_time):
+        with mock.patch("app.main.views.services.render_template") as render_template:
+            render_template.return_value = ""
+
+            self.client.get(f"/suppliers/frameworks/{framework}/submissions/{lot}/1/edit/{form}")
+
+            assert render_template.call_args[1]["session_timeout"] == display_time
```

Comment:
Do we need this extra test module with all the setup/fixtures? I was intending to just add a simple test with a `freeze_time` and an `xpath` assert to the existing test classes for the two views (i.e. `TestSupplierDeclaration` and `TestEditDraftService`).

Reply:
I was trying to avoid having the tests for the change be spread across two different files; plus the tests in `TestEditDraftService` scare me a little...

Reply:
Well ok, but it probably makes sense to stick to the convention in this app for having one test module for one code module (so we can easily find all the tests relating to a view).

Reply:
Fair point, I will split them up

Reply:
Just a simple `self.client.get` for one of the edit sections, with a `freeze_time` context manager should do the trick. We don't care about the actual form content.

Reply:
Okay I've gotten rid of that file and added simpler tests to the classes you suggested